File: templates/ruby/kderubyapp/src/%{APPNAMELC}view.rb
require 'ui_%{APPNAMELC}view_base.rb'

class %{APPNAMEFU}View < Qt::Widget
  signals 'signalChangeStatusbar(QString)', 'signalChangeCaption(QString)'
  slots :switchColors, :settingsChanged

  def initialize( parent )
    super(parent)
    @ui = Ui_%{APPNAMEFU}view_base.new
    @ui.setupUi(self)
    settingsChanged()
    setAutoFillBackground(true)
  end

private
  def switchColors()
#     # switch the foreground/background colors of the label
#     color = Settings.col_background
#     Settings.setCol_background( Settings.col_foreground )
#     Settings.setCol_foreground( color )
#     settingsChanged()
  end

  def settingsChanged()
#     pal = Qt::Palette.new
#     pal.setColor(Qt::Palette::Window, Settings::col_background)
#     pal.setColor( Qt::Palette::WindowText, Settings::col_foreground)
#     @ui.kcfg_sillyLabel.palette = pal
# 
#     # i18n : internationalization
#     @ui.kcfg_sillyLabel.text = KDE::i18n("This project is %%1 days old", Settings.val_time()) 
#     emit signalChangeStatusbar( KDE::i18n("Settings changed") )
  end
end

File: src/templates/C++/kde-frameworks5/src/%{APPNAMELC}view.cpp
/*
    SPDX-FileCopyrightText: %{CURRENT_YEAR} %{AUTHOR} <%{EMAIL}>

    SPDX-License-Identifier: GPL-2.0-only OR GPL-3.0-only OR LicenseRef-KDE-Accepted-GPL
*/

// application headers
#include "%{APPNAMELC}view.h"

#include "%{APPNAME}Settings.h"
#include "%{APPNAMELC}debug.h"


%{APPNAME}View::%{APPNAME}View(QWidget *parent)
    : QWidget(parent)
{
    m_ui.setupUi(this);
    handleSettingsChanged();
}

%{APPNAME}View::~%{APPNAME}View()
{
}

void %{APPNAME}View::switchColors()
{
    // switch the foreground/background colors of the label
    QColor color = %{APPNAME}Settings::colorBackground();
    %{APPNAME}Settings::setColorBackground(%{APPNAME}Settings::colorForeground());
    %{APPNAME}Settings::setColorForeground(color);

    handleSettingsChanged();
}

void %{APPNAME}View::handleSettingsChanged()
{
    qCDebug(%{APPNAMEUC}) << "%{APPNAME}View::handleSettingsChanged()";
    QPalette palette = m_ui.templateLabel->palette();
    palette.setColor(QPalette::Window, %{APPNAME}Settings::colorBackground());
    palette.setColor(QPalette::WindowText, %{APPNAME}Settings::colorForeground());
    m_ui.templateLabel->setPalette(palette);

    // i18n : internationalization
    m_ui.templateLabel->setText(i18n("This project is %1 days old", %{APPNAME}Settings::ageInDays()));
}

